lib/remote/httpserverconnection.cpp

```diff
@@ -171,6 +171,40 @@ void HttpServerConnection::ProcessMessageAsync(HttpRequest& request)
 
 	HttpResponse response(m_Stream, request);
 
+	Array::Ptr headerAllowOrigin = ApiListener::GetInstance()->GetAccessControlAllowOrigin();
+
+	if (headerAllowOrigin->GetLength() != 0)
+	{
+		String origin = request.Headers->Get("origin");
+
+		ObjectLock olock(headerAllowOrigin);
+
+		for (String allowedOrigin : headerAllowOrigin)
+			if (allowedOrigin == origin)
+				response.AddHeader("Access-Control-Allow-Origin", origin);
+
+		if (ApiListener::GetInstance()->GetAccessControlAllowCredentials())
+			response.AddHeader("Access-Control-Allow-Credentials", "true");
+
+		String accessControlRequestMethodHeader = request.Headers->Get("access-control-request-method");
+
+		if (!accessControlRequestMethodHeader.IsEmpty())
+		{
+			response.SetStatus(200, "OK");
+
+			response.AddHeader("Access-Control-Allow-Methods", ApiListener::GetInstance()->GetAccessControlAllowMethods());
+			response.AddHeader("Access-Control-Allow-Headers", ApiListener::GetInstance()->GetAccessControlAllowHeaders());
+
+			String msg = "<h1>Preflight OK!</h1>";
```

Comment:
This should be either JSON (if the `Accept: application/json` header is sent) or HTML.